data_validation/config_manager.py

```diff
@@ -586,12 +586,12 @@ def add_rstrip_to_comp_fields(self, comparison_fields: List[str]) -> List[str]:
 
             if (
                 source_ibis_type.is_string() or target_ibis_type.is_string()
-            ) and self._comp_field_cast(
-                # Do not add rstrip if the column is a bool hiding in a string.
+            ) and not self._comp_field_cast(
+                # Do not add rstrip if the column is a bool or UUID hiding in a string.
```

Comment:
In the PR's description you said Teradata wasn't considered, but by looking at the change you did to this condition I realized the message logged inside of it contains a Teradata description hardcoded. Just an observation, I'm not sure where exactly this part is used by the new changes.

Reply:
`self._comp_field_cast` will only return a value for a bool or uuid column. So my change above is simply updating the comment to include both data types. We could have a PostgreSQL uuid to Teradata "uuid-inside-a-string" validation. We don't need to follow the rstrip code path in those cases.

Reply:
I also thought it could apply to other engines but just took a look and turns out it is Teradata only:
```
    # As per #1190, add rstrip for Teradata string comparison fields
    if (
        config_manager.source_client.name == "teradata"
        or config_manager.target_client.name == "teradata"
    ):
        comparison_fields = config_manager.add_rstrip_to_comp_fields(comparison_fields)
```

Reply:
Ahhh that makes sense, all good here then, thanks much for confirming Neil! Resolving this topic as well